src/integration-test/java/com/commercetools/sync/integration/producttypes/ProductTypeSyncIT.java

```diff
@@ -0,0 +1,827 @@
+        final List<ProductTypeDraft> productTypeDrafts = IntStream
+            .range(0, 100)
+            .mapToObj(i -> ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "product_type_key_" + Integer.toString(i),
+                "product_type_name_" + Integer.toString(i),
+                "product_type_description_" + Integer.toString(i),
+                Arrays.asList(ATTRIBUTE_DEFINITION_DRAFT_1)
+            ))
+            .collect(Collectors.toList());
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .build();
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(productTypeSyncOptions);
+
+        final ProductTypeSyncStatistics productTypeSyncStatistics = productTypeSync
+            .sync(productTypeDrafts)
+            .toCompletableFuture().join();
+
+        AssertionsForStatistics.assertThat(productTypeSyncStatistics).hasValues(100, 100, 0, 0);
+    }
+
+
+    @Test
+    public void sync_beforeCreate_ShouldCallBeforeCreateCallback() {
+        final ProductTypeDraft newProductTypeDraft = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            PRODUCT_TYPE_KEY_2,
+            PRODUCT_TYPE_NAME_2,
+            PRODUCT_TYPE_DESCRIPTION_2,
+            singletonList(ATTRIBUTE_DEFINITION_DRAFT_1)
+        );
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .build();
+
+        ProductTypeSyncOptions spyProductTypeSyncOptions = spy(productTypeSyncOptions);
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(spyProductTypeSyncOptions);
+
+        productTypeSync.sync(singletonList(newProductTypeDraft)).toCompletableFuture().join();
+
+        verify(spyProductTypeSyncOptions).applyBeforeCreateCallBack(newProductTypeDraft);
+    }
+
+    @Test
+    public void sync_beforeCreate_ShouldNotCallBeforeUpdateCallback() {
+        final ProductTypeDraft newProductTypeDraft = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            PRODUCT_TYPE_KEY_2,
+            PRODUCT_TYPE_NAME_2,
+            PRODUCT_TYPE_DESCRIPTION_2,
+            singletonList(ATTRIBUTE_DEFINITION_DRAFT_1)
+        );
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .build();
+
+        ProductTypeSyncOptions spyProductTypeSyncOptions = spy(productTypeSyncOptions);
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(spyProductTypeSyncOptions);
+
+        productTypeSync.sync(singletonList(newProductTypeDraft)).toCompletableFuture().join();
+
+        verify(spyProductTypeSyncOptions, never()).applyBeforeUpdateCallBack(any(), any(), any());
+    }
+
+    @Test
+    public void sync_beforeUpdate_ShouldCallBeforeUpdateCallback() {
```

Comment:
I am wondering why are these tests significant? Namely: 

- public void sync_beforeCreate_ShouldCallBeforeCreateCallback
- public void sync_beforeCreate_ShouldNotCallBeforeUpdateCallback
- public void sync_beforeUpdate_ShouldCallBeforeUpdateCallback
- public void sync_beforeUpdate_ShouldNotCallBeforeCreateCallback

Reply:
Basically is to verify that the right callbacks are called in the correct stages. 

i.e How would you verify that `applyBeforeUpdateCallBack` function is called when a product type is created?

Reply:
but do you think we really need an integration test for that, isn't a unit test sufficient to test that behaviour?